File: scripts/create-import-example.ts
#!/usr/bin/env node
/**
 * Script to create an Excel file example for importing classes data.
 * Uses the xlsx library that's already installed in the project.
 */

import * as XLSX from 'xlsx';
import * as fs from 'fs';
import * as path from 'path';

function createClassesImportExample() {
  // Define the data structure
  const data = [
    {
      'ID класса': 'class-001',
      'Название класса': '5А',
      'Количество учеников': 25,
      'Ответственный учитель ID': 'teacher-001',
      'Ответственный учитель ФИО': 'Иванов Иван Иванович'
    },
    {
      'ID класса': 'class-002',
      'Название класса': '6Б',
      'Количество учеников': 28,
      'Ответственный учитель ID': 'teacher-002',
      'Ответственный учитель ФИО': 'Петрова Мария Сергеевна'
    },
    {
      'ID класса': 'class-003',
      'Название класса': '7В',
      'Количество учеников': 23,
      'Ответственный учитель ID': 'teacher-003',
      'Ответственный учитель ФИО': 'Сидоров Алексей Петрович'
    },
    {
      'ID класса': 'class-004',
      'Название класса': '8Г',
      'Количество учеников': 26,
      'Ответственный учитель ID': 'teacher-004',
      'Ответственный учитель ФИО': 'Козлова Елена Владимировна'
    }
  ];

  // Create a new workbook
  const workbook = XLSX.utils.book_new();
  
  // Convert data to worksheet
  const worksheet = XLSX.utils.json_to_sheet(data);
  
  // Set column widths
  worksheet['!cols'] = [
    { wch: 15 },  // ID класса
    { wch: 18 },  // Название класса
    { wch: 23 },  // Количество учеников
    { wch: 28 },  // Ответственный учитель ID
    { wch: 35 }   // Ответственный учитель ФИО
  ];
  
  // Add the worksheet to the workbook
  XLSX.utils.book_append_sheet(workbook, worksheet, 'Классы');
  
  // Ensure the public directory exists
  const outputDir = path.join(process.cwd(), 'public');
  if (!fs.existsSync(outputDir)) {
    fs.mkdirSync(outputDir, { recursive: true });
  }
  
  // Write the file
  const outputPath = path.join(outputDir, 'import_example.xlsx');
  XLSX.writeFile(workbook, outputPath);
  
  console.log('✓ Excel file created successfully!');
  console.log(`✓ Location: ${outputPath}`);
  console.log('✓ Sheet name: Классы');
  console.log(`✓ Number of example rows: ${data.length}`);
  console.log('✓ Columns: ID класса, Название класса, Количество учеников, Ответственный учитель ID, Ответственный учитель ФИО');
}

// Run the script
try {
  createClassesImportExample();
} catch (error) {
  console.error('Error creating Excel file:', error);
  process.exit(1);
}

File: scripts/create_import_example.py
#!/usr/bin/env python3
"""
Script to create an Excel file example for importing classes data.
"""

from openpyxl import Workbook
from openpyxl.styles import Font, PatternFill, Alignment
import os

def create_classes_import_example():
    """Create an Excel file with example classes data."""
    
    # Create a new workbook and select the active sheet
    wb = Workbook()
    ws = wb.active
    ws.title = "Классы"
    
    # Define headers
    headers = [
        "ID класса",
        "Название класса",
        "Количество учеников",
        "Ответственный учитель ID",
        "Ответственный учитель ФИО"
    ]
    
    # Add headers with styling
    header_fill = PatternFill(start_color="4472C4", end_color="4472C4", fill_type="solid")
    header_font = Font(bold=True, color="FFFFFF")
    
    for col_num, header in enumerate(headers, 1):
        cell = ws.cell(row=1, column=col_num)
        cell.value = header
        cell.fill = header_fill
        cell.font = header_font
        cell.alignment = Alignment(horizontal="center", vertical="center")
    
    # Example data
    example_data = [
        ["class-001", "5А", 25, "teacher-001", "Иванов Иван Иванович"],
        ["class-002", "6Б", 28, "teacher-002", "Петрова Мария Сергеевна"],
        ["class-003", "7В", 23, "teacher-003", "Сидоров Алексей Петрович"],
        ["class-004", "8Г", 26, "teacher-004", "Козлова Елена Владимировна"],
    ]
    
    # Add example data
    for row_num, row_data in enumerate(example_data, 2):
        for col_num, value in enumerate(row_data, 1):
            cell = ws.cell(row=row_num, column=col_num)
            cell.value = value
            cell.alignment = Alignment(horizontal="left" if col_num != 3 else "center", vertical="center")
    
    # Adjust column widths
    column_widths = [18, 20, 25, 30, 35]
    for col_num, width in enumerate(column_widths, 1):
        ws.column_dimensions[chr(64 + col_num)].width = width
    
    # Ensure the public directory exists
    output_dir = "public"
    os.makedirs(output_dir, exist_ok=True)
    
    # Save the file
    output_path = os.path.join(output_dir, "import_example.xlsx")
    wb.save(output_path)
    
    print(f"✓ Excel file created successfully at: {output_path}")
    print(f"✓ Sheet name: {ws.title}")
    print(f"✓ Number of example rows: {len(example_data)}")
    print(f"✓ Columns: {', '.join(headers)}")

if __name__ == "__main__":
    create_classes_import_example()
